### ISSUE TITLE
[exporter/signalfx] Permanent error is not returned in non-retryable cases

### ISSUE THREAD

**Issue by jrcamp** [open-telemetry/opentelemetry-collector-contrib#1495]
When the server returns 400 Bad Request we retry sending even though it will likely be rejected each time:

```
{"level":"info","ts":1604093487.2966263,"caller":"exporterhelper/queued_retry.go:252","msg":"Exporting failed. Will retry the request after interval.","component_kind":"exporter","component_type":"signalfx","component_name":"signalfx","error":"HTTP 400 \"Bad Request\"","interval":"4.646742427s"}
```

Needs some digging to see if it's only 400 that should be returned as permanent or if there are others. (Or maybe just all 4xx)

**Comment by flands** [comment]
Can't be all 4xx right? `429` should retry.

**Comment by rmfitzpatrick** [comment]
If we set any error as permanent, the combined error will be deemed permanent and all retries are prevented.  I think we need to implement a partial error and the associated handler in the queued processor: https://github.com/open-telemetry/opentelemetry-collector/blob/8ceddba7ea03216b077582bab5d0e949a3297f5b/processor/queuedprocessor/queued_processor.go#L152

**Comment by bogdandrutu** [comment]
PR #1672 will partially fix this, because we need a way to group requests by access token before adding to the queue, so that we can send the entire data in one request, otherwise we need to return permanent error for some data, throttle error for others, and maybe we can use partial error but that may complicate things.

For the moment just return the error with the idea that it is only one ResourceMetrics per message which is the case when receiving data using SignalFx receiver and no batching.

**Comment by jrcamp** [comment]
This was fixed in #1672
